### PR TITLE
govc: Support import.{ova,ovf,spec} a URL

### LINKED ISSUE #651: URL supports for ova.import
I tried the latest govc CLI to upload an OVA at one web site with certain URL, but not working for me. Seem ova file has to be on the same machine as govc CLI.

Deploy ova from URL is supported in standard vsphere cli and web application. Can we have the same feature here?

thanks
-weiwen

### PR DESCRIPTION
**Pull request by anfernee**
Fixes #651

### CODE REVIEW

File: govc/importx/archive.go [hunk 1]
```diff
@@ -127,23 +132,109 @@ func (t *TapeArchive) Open(name string) (io.ReadCloser, int64, error) {
...
+	cache := cachePath(fpath)
+	_, err := os.Stat(cache)
+	if err == nil && !d.nocache {
+		return cache, true, nil
+	}
+
+	return cache, false, d.download(fpath, cache)
+}
+
+func (d Downloader) download(remote, local string) error {
```
Comment (dougm) [inline govc/importx/archive.go]: Can we re-use `soap.Client.DownloadFile` instead?

File: govc/importx/archive.go [hunk 2]
```diff
@@ -127,23 +132,109 @@ func (t *TapeArchive) Open(name string) (io.ReadCloser, int64, error) {
...
+
+	_, err = io.Copy(f, r)
+
+	return err
+}
+
+var cacheDir = "/tmp/govc-cache/"
+
+func init() {
+	_ = os.MkdirAll(cacheDir, 0755)
```
[the remark this answers is not in the document]
Comment (dougm) [reply govc/importx/archive.go]: Actually, I don't think we should do any caching at all.  If users want that, they can use curl or wget.  We can stream everything without ever hitting the disk and it's ok if we can only support .ovf (not .ova) for this case.

File: govc/importx/archive.go (hunk 1 above)
Comment (anfernee) [reply govc/importx/archive.go]: sure.

File: govc/importx/archive.go [hunk 3]
```diff
@@ -127,23 +132,109 @@ func (t *TapeArchive) Open(name string) (io.ReadCloser, int64, error) {
...
+			sinker.(progress.Waiter).Wait()
+		}()
+	}
+
+	_, err = io.Copy(f, r)
+
+	return err
+}
+
+var cacheDir = "/tmp/govc-cache/"
```
[the remark this answers is not in the document]
Comment (anfernee) [reply govc/importx/archive.go]: agreed.

File: govc/importx/archive.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (anfernee) [reply govc/importx/archive.go]: I thought about direct stream too. but when I look at the ova files which is almost always 2G size, I feel cache is useful. I have a `nocache` option in Downloader, but haven't linked it with flags. Do you think we can use that?

**Comment by anfernee** [comment]
Let's follow what ovftool do without caching. Use solution #886

**Pull request closed without merging**